network/router.go

```diff
@@ -40,6 +40,11 @@ type Router struct {
 
 	// wg waits for all handleConn routines to be done.
 	wg sync.WaitGroup
+
+	// Every handler in this list is called by this router when any network error occurs (Timeout, Connection
+	// Closed, or EOF). Those handler should be added by using SetErrorHandler(). The 1st argument is the remote
+	// server with whom the error happened
+	errorHandlers []func(*ServerIdentity)
```

Comment:
Please, change the name to reflect exactly the underlying meaning (here and in every other references).
> `any network error occurs `

That's wrong, there could be a listener error, there could be a file limit exhaustion error etc etc. 
`ConnDropped` or something else explicit.